qiskit/transpiler/passes/commutation_analysis.py

```diff
@@ -88,156 +93,36 @@ def run(self, dag):
-
-        if node.name in ['cx', 'cy', 'cz']:
-
-            qstate_list_ext = [np.identity(2)] * wire_num
-
-            node_ctrl = "{0}[{1}]".format(str(node.qargs[0][0].name), str(node.qargs[0][1]))
-            node_tgt = "{0}[{1}]".format(str(node.qargs[1][0].name), str(node.qargs[1][1]))
-            ctrl = wires.index(node_ctrl)
-            tgt = wires.index(node_tgt)
-
-            qstate_list[ctrl] = _gate_master_def(name='P0')
-            qstate_list[tgt] = _gate_master_def(name='Id')
-            qstate_list_ext[ctrl] = _gate_master_def(name='P1')
-            if node.name == 'cx':
-                qstate_list_ext[tgt] = _gate_master_def(name='x')
-            if node.name == 'cy':
-                qstate_list_ext[tgt] = _gate_master_def(name='y')
-            if node.name == 'cz':
-                qstate_list_ext[tgt] = _gate_master_def(name='z')
-
-            rt_list = [qstate_list] + [qstate_list_ext]
-
-        else:
-
-            mat = _gate_master_def(name=node.name, params=node.op.params)
-            node_num = "{0}[{1}]".format(str(node.qargs[0][0].name),
-                                         str(node.qargs[0][1]))
-            qstate_list[wires.index(node_num)] = mat
-
-            rt_list = [qstate_list]
+def _commute(node1, node2):
 
-        crt = np.zeros([2 ** wire_num, 2 ** wire_num])
+    if node1.type != "op" or node2.type != "op":
+        return False
 
-        for state in rt_list:
-            crt = crt + _kron_list(state)
+    new_qreg = []
 
-        final_unitary = np.dot(crt, final_unitary)
-    return final_unitary
+    for node in [node1, node2]:
+        for wire in node.qargs:
+            if wire not in new_qreg:
+                new_qreg.append(wire)
 
+    new_qr = QuantumRegister(len(new_qreg))
 
-def _kron_list(args):
-    ret = args[0]
-    for item in args[1:]:
-        ret = np.kron(ret, item)
-    return ret
+    circ_n1n2 = QuantumCircuit(new_qr)
+    circ_n2n1 = QuantumCircuit(new_qr)
 
+    for node in [node1, node2]:
+        qarg_list = []
+        for wire in node.qargs:
+            qarg_list.append(new_qr[new_qreg.index(wire)])
+        circ_n1n2.append(node.op, qargs=qarg_list)
 
-def _matrix_commute(node1, node2):
-    # Good for composite gates or any future
-    # user-defined gate of equal or less than 2 qubits.
-    ret = False
-    if set(node1.qargs) & set(node2.qargs) == set():
-        ret = True
-    if _calc_product(node1, node2) is not None:
-        ret = np.allclose(_calc_product(node1, node2),
-                          _calc_product(node2, node1),
-                          atol=_CUTOFF_PRECISION)
-    return ret
+    for node in [node2, node1]:
+        qarg_list = []
+        for wire in node.qargs:
+            qarg_list.append(new_qr[new_qreg.index(wire)])
+        circ_n2n1.append(node.op, qargs=qarg_list)
 
+    if_commute = np.allclose(Operator(circ_n1n2).data, Operator(circ_n2n1).data,
+                             atol=_CUTOFF_PRECISION)
```

Comment:
Instead of creating mini-circuits, then simulating them to create operators, then seeing if the matrices associated with those operators commute, I think this whole block of code can be simplified:
- first, don't use the raw data. the `commutator` is defined so that it can be used here.
- second, the Operator class already has a `compose` method which I think can be used here. This method takes a `qargs` argument, which specifies how a second operator should be attached to the first operator. So that's basically like creating a circuit, except you don't need all these registers.